### PR TITLE
Fixes #5112 (Fix error messages raised when multiple modules have the same original name)

### LINKED ISSUE #5112: Module aliasing fails when multiple modules have the same original name
1. Create two empty Nim source files:

```bash
touch a.nim
mkdir stuff
touch stuff/a.nim
```

2. Create a main Nim source that imports these two modules:

```nim
import a
#import stuff/a  # Error: redefinition of 'a'  -- CORRECT
import stuff/a as b   # should work, but fails with linker error -- INCORRECT
```

3. Attempting to compile main.nim will result in a linker error:

```
/media/sf_D_DRIVE/Work/Code/nim-lang-bugs/import-bug/nimcache/a.o: In function `unknown_aInit000':
a.c:(.text+0x97): multiple definition of `unknown_aInit000'
/media/sf_D_DRIVE/Work/Code/nim-lang-bugs/import-bug/nimcache/a.o:a.c:(.text+0x97): first defined here
/media/sf_D_DRIVE/Work/Code/nim-lang-bugs/import-bug/nimcache/a.o: In function `unknown_aDatInit000':
a.c:(.text+0xd0): multiple definition of `unknown_aDatInit000'
/media/sf_D_DRIVE/Work/Code/nim-lang-bugs/import-bug/nimcache/a.o:a.c:(.text+0xd0): first defined here
collect2: error: ld returned 1 exit status
Error: execution of an external program failed: 'gcc   -o /media/sf_D_DRIVE/Work/Code/nim-lang-bugs/import-bug/main  /media/sf_D_DRIVE/Work/Code/nim-lang-bugs/import-bug/nimcache/a.o /media/sf_D_DRIVE/Work/Code/nim-lang-bugs/import-bug/nimcache/a.o /media/sf_D_DRIVE/Work/Code/nim-lang-bugs/import-bug/nimcache/stdlib_system.o /media/sf_D_DRIVE/Work/Code/nim-lang-bugs/import-bug/nimcache/main.o  -ldl'
```

### PR DESCRIPTION
**Pull request by loloiccl**
Fix error messages raised when multiple modules have the same original name as the following.

files:
    
    a.nim
    a2.nim
    b.nim (empty)
    dir/b.nim (empty)
    
a.nim:
    
    import b
    import dir/b as c    

a2.nim
    
    import b
    import dir/b

 compile:
    
    nim c a.nim
    
raise error:
    
    Error: module names need to be unique per Nimble package; ...
    
compile:
    
    nim c a2.nim
    
raise error:
    
    Error: module names need to be unique per Nimble package; ...

### CODE REVIEW

**Comment by Araq** [comment]
Fails tester with:

compiler/options.nim(1, 2) Error: module names need to be unique per Nimble package; module clashes with /home/travis/build/nim-lang/Nim/dist/nimble/src/nimblepkg/options.nim

And indeed the patch looks quite wrong, you must not check uniqueness against the full graph.

**Comment by loloiccl** [comment]
Sorry I confuse what is the project.
I pushed the new patch.

File: compiler/modules.nim [hunk 1]
```diff
@@ -133,11 +133,12 @@ proc newModule(graph: ModuleGraph; fileIdx: int32): PSym =
     rawMessage(errInvalidModuleName, result.name.s)
 
   result.info = newLineInfo(fileIdx, 1, 1)
-  let pack = getIdent(getPackageName(filename))
+  let
+    pck = getPackageName(filename)
+    pck2 = if pck.len > 0: pck else: "unknown-package"
```
Comment (Araq) [inline compiler/modules.nim]: why change this to 'unknown-package'? Please don't. These names might end up in the name hashes resulting in huge recompile times.

Comment (loloiccl) [inline compiler/modules.nim]: Sorry. I'll fix soon.

I think invalid name for package is better than valid one for this.
What do you think?
If invalid name is better, I will register this as another issue.

**Pull request merged**